### PR TITLE
RDS & RData IO modules

### LINKED ISSUE #12: Modules from_rdata and to_rdata

### PR DESCRIPTION
**Pull request by athityakumar**
Closes #12 

This Pull Request adds Importers & Exporters for `.rds`, `.RData` file formats. The RDS Importer, RDS Exporter and RData Importer are monkey-patched into `Daru::DataFrame` like other IO modules. Whereas, the RData exporter would fare well as a stand-alone feature IMO (like `Daru::IO::Exporters::RData.new('file.RData', 'first.df': df1, 'second.df': df2, ...).call`.

YARD Docs have been added. Tests are currently added for just 1 just fixture file each for both the Importers, and more tests will be added in this PR soon. After they have been added, this PR will be ready for a review.

### CODE REVIEW

File: lib/daru/io/exporters/rds.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        private
+
+        def process_statements(r_variable, dataframe)
+          [].tap do |statement|
+            dataframe.map_vectors_with_index do |vector, i|
+              statement << "#{i} = c(#{vector.to_a.map { |val| convert_datatype(val) }.join(', ')})"
+            end
+            statement << "#{r_variable} = data.frame(#{dataframe.vectors.to_a.map(&:to_s).join(', ')})"
+          end
```
Comment (zverok) [inline lib/daru/io/exporters/rds.rb]: I'd say 
```ruby
[
  *dataframe.map_vectors_with_index { |vector, i| "#{i} = c(#{vector_statement(vector)})" }
  "#{r_variable} = data.frame(#{dataframe.vectors.to_a.map(&:to_s).join(', ')})"
]
```
...or something like this, WDYT?..

File: lib/daru/io/exporters/rds.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        end
+
+        def convert_datatype(value)
+          return 'NA' unless value
+          return value unless value.is_a?(String)
+          "'#{value}'"
```
Comment (zverok) [inline lib/daru/io/exporters/rds.rb]: For me it is a real good case for `case`, sorry for the pun.

File: lib/daru/io/importers/rds.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        def process_dataframe(data)
+          data = data.map { |key, values| [key.to_sym, values.map { |val| convert_datatype(val) }] }.to_h
+          Daru::DataFrame.new(data)
+        end
+
+        def convert_datatype(value)
+          return nil if value.is_a?(Float) && value.nan?
+          return value.to_f if value == value.to_f.to_s
+          return value.to_i if value == value.to_i.to_s
```
Comment (zverok) [inline lib/daru/io/importers/rds.rb]: I am not sure this code is clean enough. (For ex, there could be slight difference in string representation of floats in R and Ruby). Don't we have other options?.. Like check it with regexp. And use `case` again btw ;)

File: lib/daru/io/exporters/rds.rb (hunk 1 above)
Comment (athityakumar) [inline lib/daru/io/exporters/rds.rb]: This makes it more readable for the RDS Exporter, but a similar approach to RData Exporter doesn't seem more readable (to me)  than the current `[].tap`. And I wouldn't prefer to look at different semantics of `process_statements` method in RDS & RData exporters.

Comment (athityakumar) [inline lib/daru/io/exporters/rds.rb]: Oops, very sorry. While typing the above reply, I was looking at a previous commit where RData Exporter hadn't inherited RDS Exporter with extreme confusion on whether had time-travelled back by 2 days. :-/

Yes, this makes sense.

File: lib/daru/io/importers/rds.rb (hunk 3 above)
Comment (athityakumar) [inline lib/daru/io/importers/rds.rb]: Changed this to `case`, but I'm not sure whether switching to regexps would be more reliable && more readable.

**Pull request merged**